### PR TITLE
Ruff: Add ASYNC checks

### PR DESCRIPTION
**Pull request by cclauss**
https://docs.astral.sh/ruff/rules/#flake8-async-async
> examples/rest-api/main.py:77:22: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [ ] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by deshraj** [comment]
@cclauss thanks for the PR. Can you please resolve the conflicts so that we can review the PR? 

Thank you for your time.

**Comment by cclauss** [comment]
@deshraj Rebased.

File: pyproject.toml [hunk 1]
```diff
@@ -152,7 +152,7 @@ google-api-core = { version = "^2.15.0", optional = true }
 [tool.poetry.group.dev.dependencies]
 black = "^23.3.0"
 pre-commit = "^3.2.2"
-ruff = "^0.0.220"
+ruff = "^0.1.11"
```
Comment (deshraj) [inline pyproject.toml:155]: Any specific reason to update ruff here? I am not sure if we are ready to update the ruff version yet.

Comment (cclauss) [reply pyproject.toml:155]: `ASYNC` issues were not in v0.0.220.

If the code passes then why not keep the linter current?

Comment (deshraj) [reply pyproject.toml:155]: Thanks for the explanation. 

One more question: What command did you use to update the poetry.lock file?

Comment (cclauss) [reply pyproject.toml:155]: I am no good at Poetry so I tried three things and the fourth one actually did something!  ;-)

I would be happy to revert the changes to `poetry.lock` and then run whatever command you suggest.

Comment (deshraj) [reply pyproject.toml:155]: Yes, can you please remove the changes from the poetry.lock file?

Comment (cclauss) [reply pyproject.toml:155]: `poetry lock --no-update`

**Review by deshraj — APPROVED** [review]
Nice, thanks

**Pull request merged**